compiler/semexprs.nim

```diff
@@ -2997,6 +2998,53 @@ proc semPragmaStmt(c: PContext; n: PNode) =
   else:
     pragma(c, c.p.owner, n, stmtPragmas, true)
 
+proc resolveIdent(c: PContext, ident: PIdent, s: var PSym, n: PNode,
+                  flags: TExprFlags, expectedType: PType): PNode =
```

Comment:
So ... it returns what exactly? And when does it overwrite the `s` parameter? Convoluted.

Reply:
I swapped `s` and the result value, and made it so only n's info is needed. Every parameter seems to be needed though, not sure how to get this cleaner.

Reply:
Now merged `n`/`info` and `ident` into the same parameter (just `n`), and renamed to `resolveIdentToSym`. Hopefully this is clearer.